inbm/dispatcher-agent/dispatcher/source/ubuntu_source_manager_genai_unit_test.py

```diff
@@ -0,0 +1,368 @@
+import sys
+sys.path.insert(0, '../../')
+sys.path.insert(0, '../../../../')
+sys.path.insert(0, '../')
+sys.path.insert(0, './')
+sys.path.insert(0, '../../../')
+sys.path.insert(0, '/home/runner/GITHUB_ACTION_RUNNERS/_work/intel-inb-manageability/intel-inb-manageability/inbm/dispatcher-agent/dispatcher/source/')
+sys.path.insert(0, '/home/runner/GITHUB_ACTION_RUNNERS/_work/intel-inb-manageability/intel-inb-manageability/inbm-lib/')
+sys.path.insert(0, '/home/runner/GITHUB_ACTION_RUNNERS/_work/intel-inb-manageability/intel-inb-manageability/inbm/dispatcher-agent/')
+import pytest
+from unittest.mock import patch, MagicMock
+from dispatcher.source.ubuntu_source_manager import UbuntuApplicationSourceManager, ApplicationAddSourceParameters
+from unittest.mock import patch, mock_open
+from dispatcher.source.ubuntu_source_manager import UbuntuApplicationSourceManager, ApplicationSourceList
+from dispatcher.source.ubuntu_source_manager import UbuntuApplicationSourceManager, ApplicationRemoveSourceParameters, SourceError, LINUX_GPG_KEY_PATH, UBUNTU_APT_SOURCES_LIST_D
+from dispatcher.source.ubuntu_source_manager import UbuntuApplicationSourceManager, SourceError, ApplicationUpdateSourceParameters
+from dispatcher.source.ubuntu_source_manager import UbuntuOsSourceManager, SourceParameters, SourceError
+from dispatcher.source.ubuntu_source_manager import UbuntuOsSourceManager, SourceError
+from unittest.mock import patch, mock_open, call
+
+#DO NOT DELETE THIS LINE - TestUbuntuApplicationSourceManagerAdd
+'''
+ADD HUMAN FEEDBACK BELOW:
+
+'''
+class TestUbuntuApplicationSourceManagerAdd:
```

Comment:
To go back to Pull Request Summary table, Click on the [link](https://github.com/intel/intel-inb-manageability/pull/467#)

<h3>Class: UbuntuApplicationSourceManager |Function: add</h3>

[Source Code Link](https://github.com/intel/intel-inb-manageability/blob/NEX_MAXIMUS_UT/develop/inbm/dispatcher-agent/dispatcher/source/ubuntu_source_manager.py?plain=1#L103)

[Unit Test Case Code Link](https://github.com/intel/intel-inb-manageability/blob/NEX_MAXIMUS_UT/develop/inbm/dispatcher-agent/dispatcher/source/ubuntu_source_manager_genai_unit_test.py?plain=1#L21)

<h4>Test Case Scenarios:</h4>
The unit test `TestUbuntuApplicationSourceManagerAdd` generates two test cases:

1. Test Case: `test_add('key1', 'http://example.com/key1', 'file1', ['deb http://example.com/ubuntu/ bionic main restricted'])`

   This test case tests the scenario where a GPG key name and URI are provided along with the source list file name and sources. The `add` method is expected to verify the GPG key URI, add the GPG key, and create a file with the provided sources. The test asserts that the `verify_source`, `add_gpg_key`, and `create_file_with_contents` functions are called with the correct arguments.

2. Test Case: `test_add(None, None, 'file2', ['deb http://example.com/ubuntu/ bionic main restricted'])`

   This test case tests the scenario where no GPG key name and URI are provided, only the source list file name and sources. The `add` method is expected to create a file with the provided sources. The test asserts that the `create_file_with_contents` function is called with the correct arguments and that the `verify_source` and `add_gpg_key` functions are not called.

In both test cases, the `add` method is called with an instance of `ApplicationAddSourceParameters` as the argument. The `ApplicationAddSourceParameters` instance is created with the test parameters. The `add` method is part of the `UbuntuApplicationSourceManager` class, which is mocked in the test. The `DispatcherBroker` class, which is a dependency of the `UbuntuApplicationSourceManager` class, is also mocked.


<h4>Test Execution Results:</h4>

| Total Cases    | 2  |
| -------------- | -- |
| Passed Cases   | 2  |
| Failed Cases   | 0  |



To go back to Pull Request Summary table, Click on the [link](https://github.com/intel/intel-inb-manageability/pull/467#)